packages/fx-core/src/component/generator/generatorAction.ts

```diff
@@ -4,28 +4,64 @@
 import AdmZip from "adm-zip";
 import path from "path";
 import { GeneratorContext } from "./generatorContext";
-import { fetchZipFromUrl, fetchTemplateZipUrl, unzip, fetchSampleZipUrl } from "./utils";
+import { fetchZipFromUrl, fetchTemplateZipUrl, unzip, fetchSampleZipUrl, zipFolder } from "./utils";
 import fs from "fs-extra";
 import { defaultTimeoutInMs, defaultTryLimits } from "./constant";
 import { getTemplatesFolder } from "../../folder";
 import { MissKeyError } from "./error";
+import { FeatureFlagName } from "../../common/constants";
 
 export interface GeneratorAction {
   name: string;
   run: (context: GeneratorContext) => Promise<void>;
 }
 
 export enum GeneratorActionName {
+  FetchTemplateZipFromSourceCode = "FetchTemplateZipFromSourceCodeAction",
   FetchTemplateUrlWithTag = "FetchTemplatesUrlWithTag",
   FetchSampleUrlWithTag = "FetchSamplesUrlWithTag",
   FetchZipFromUrl = "FetchZipFromUrl",
   FetchTemplateZipFromLocal = "FetchTemplateZipFromLocal",
   Unzip = "Unzip",
 }
 
+// * This action is only for debug purpose
```

Comment:
This action is for debug. Will the interface be different from user scenario?

Reply:
No. This is nothing about interface. By default, generator gets template from remote releases. When setting `FeatureFlagName.DebugTemplate` to true and F5 from vscode extension, it gets templates from local source code rather than remote releases.
You can get more details from [how-to-debug-templates](https://github.com/OfficeDev/TeamsFx/blob/dev/templates/CONTRIBUTING.md#how-to-debug-templates)